qiskit/algorithms/minimum_eigensolvers/numpy_minimum_eigensolver.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The NumPy minimum eigensolver algorithm and result."""
+
+from __future__ import annotations
+
+from typing import Callable
+import logging
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+# TODO this path will need updating
+from ..eigen_solvers.numpy_eigen_solver import NumPyEigensolver
+from .minimum_eigensolver import MinimumEigensolver, MinimumEigensolverResult
+from ..list_or_dict import ListOrDict
+
+logger = logging.getLogger(__name__)
+
+
+class NumPyMinimumEigensolver(MinimumEigensolver):
+    """
+    The NumPy minimum eigensolver algorithm.
+    """
+
+    def __init__(
+        self,
+        filter_criterion: Callable[
+            [list | np.ndarray, float, ListOrDict[float] | None], bool
+        ] = None,
```

Comment:
I'm not sure whether it requires `| None`?
```suggestion
        ] | None = None,
```

Reply:
The prior signature was
>        filter_criterion: Callable[[Union[List, np.ndarray], float, Optional[ListOrDict[float]]], bool] = None,
 
 I imagine this is saying that the filter criteria is applied to the aux ops result which may be None. I would have thought the filter would not have needed to be called in this case but this is how it has been.

Reply:
Change made in the commit that addresses your comment below.